CONTRIBUTING.md

```diff
@@ -5,33 +5,44 @@ meeting](https://github.com/open-telemetry/community#special-interest-groups).
+1. *End-users*, aiming to use an OpenTelemetry Collector binary.
+1. *Component developers*, consuming the Go APIs to create components compatible with the OpenTelemetry Collector Builder.
+1. *Collector library users*, consuming other Go APIs exposed by the opentelemetry-collector repository, for example to
+   build custom distributions or other projects building on top of the Collector Go APIs.
+
+When the needs of these audiences conflict, end-users should be prioritized, followed by component developers, and
+finally Collector library users.
 
 ### End-users
 
 End-users are the target audience for our binary distributions, as made available via the
-[opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases) repository. To
-them, stability in the behavior is important, be it runtime or configuration. They are more numerous and harder to get
-in touch with, making our changes to the collector more disruptive to them than to other audiences. As a general rule,
-whenever you are developing OpenTelemetry Collector components (extensions, receivers, processors, exporters), you
-should have end-users' interests in mind. Similarly, changes to code within packages like `config` will have an impact
-on this audience. Make sure to cause minimal disruption when doing changes here.
-
-### Collector distributions
-
-In this capacity, the opentelemetry-collector repository's public Go types, functions, and interfaces act as an API for
-other projects. In addition to the end-user aspect mentioned above, this audience also cares about API compatibility,
-making them susceptible to our refactorings, even though such changes wouldn't cause any impact to end-users. See the
-"Breaking changes" in this document for more information on how to perform changes affecting this audience.
-
-This audience might use tools like the
-[opentelemetry-collector-builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder) as
-part of their delivery pipeline. Be mindful that changes there might cause disruption to this audience.
+[opentelemetry-collector-releases](https://github.com/open-telemetry/opentelemetry-collector-releases) repository, as
+well as distributions created using the [OpenTelemetry Collector
+Builder](https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder). To them, stability in the
+behavior is important, be it runtime or configuration. They are more numerous and harder to get in touch with, making
+our changes to the Collector more disruptive to them than to other audiences. As a general rule, whenever you are
+developing OpenTelemetry Collector components (extensions, receivers, processors, exporters, connectors), you should
+have end-users' interests in mind. Similarly, changes to code within packages like `config` will have an impact on this
```

Comment:
I'd like to call out changes to component telemetry also have a significant impact on these users.

Reply:
I reworded to say

> stability in the behavior is important, be it runtime, configuration or [internal telemetry](...)